app/models/service_order.rb

```diff
@@ -4,12 +4,31 @@ class ServiceOrder < ActiveRecord::Base
   has_many :miq_requests, :dependent => :nullify
 
   validates :state, :inclusion => {:in => %w(wish cart ordered)}
-  validates :name, :presence => true
   validates :state, :presence => true
+  validates :name, :presence => true, :on => :update
 
-  before_create :assign_user_name
+  before_create :assign_user_name, :assign_tenant
+  after_create  :create_order_name
 
   def assign_user_name
     self.user_name = user.try(:name)
   end
+
+  def assign_tenant
+    self.tenant = tenant
```

Comment:
This method is not doing anything in its current form.  It is either not needed or suppose to be something more like:
`self.tenant ||= user.current_tenant`

Note: You are passing the tenant in `create_service_order_and_checkout` based on the request.